src/redis/cache_impl.go

```diff
@@ -210,6 +214,27 @@ func (this *rateLimitCacheImpl) DoLimit(
 	return responseDescriptorStatuses
 }
 
+func CalculateReset(currentLimit *pb.RateLimitResponse_RateLimit) *duration.Duration {
+	sec := unitInSeconds(currentLimit.Unit)
+	now := limiter.NewTimeSourceImpl().UnixNow()
```

Comment:
Does creating a time source cost anything perf-wise? Can you pre-create the time source and/or pass it in to the constructor?